contrib/inventory/vmware_inventory.py

```diff
@@ -561,6 +571,65 @@ def create_template_mapping(self, inventory, pattern, dtype='string'):
             mapping[k] = newkey
         return mapping
 
+    def create_host_mapping(self, inventory, pattern, use_hw_if='False', ip_filter="", dtype='string'):
```

Comment:
This function is too complex. Could you please factor and avoid:
- more than 2 level of condition
- inner loop.
You can probably break down the function in several small functions.